ckanext/switzerland/helpers/dataset_form_helpers.py

```diff
@@ -49,22 +49,43 @@ def ogdch_publisher_form_helper(data):
     fills the publisher form snippet either from a previous form entry
     or from the db
     """
-    publisher_form_name = data.get('publisher-name')
+
+    # check for form inputs first
+    publisher_form_name = {
+        'fr': data.get('publisher-name-fr', ''),
+        'en': data.get('publisher-name-en', ''),
+        'de': data.get('publisher-name-de', ''),
+        'it': data.get('publisher-name-it', ''),
+    }
     publisher_form_url = data.get('publisher-url')
-    publisher_in_form = publisher_form_url or publisher_form_name
-    if publisher_in_form:
+
+    if publisher_form_url or any(publisher_form_name.values()):
         return {'name': publisher_form_name,
                 'url': publisher_form_url}
 
+    # check for publisher from db
     publisher_stored = data.get('publisher')
     if publisher_stored:
-        return json.loads(publisher_stored)
+        # handle stored publisher data (both as dict or string)
+        if isinstance(publisher_stored, dict):
+            name = publisher_stored.get('name', {})
+            log.info("publisher_stored ins")
```

Comment:
Out of interest: what does `ins` stand for?
And have you considered changing the log-level to `debug` or do you prefer having this logged every time for the beginning?

Reply:
I have deleted these logs